src/app/clusters/refrigerator-alarm-server/refrigerator-alarm-server.cpp

```diff
@@ -0,0 +1,148 @@
+/**
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ *
+ */
+#include "refrigerator-alarm-server.h"
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/cluster-enums.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/EventLogging.h>
+#include <app/InteractionModelEngine.h>
+#include <app/util/attribute-storage.h>
+#include <app/util/error-mapping.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::RefrigeratorAlarm;
+using namespace chip::app::Clusters::RefrigeratorAlarm::Attributes;
+using namespace chip::DeviceLayer;
+using chip::Protocols::InteractionModel::Status;
+
+using namespace std;
+RefrigeratorAlarmServer RefrigeratorAlarmServer::instance;
+
+RefrigeratorAlarmServer & RefrigeratorAlarmServer::Instance()
+{
+    return instance;
+}
+
+EmberAfStatus RefrigeratorAlarmServer::GetMaskValue(EndpointId endpoint, BitMask<AlarmMap> * mask)
+{
+    EmberAfStatus status = Attributes::Mask::Get(endpoint, mask);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        ChipLogProgress(Zcl, "Refrigerator Alarm: ERR: reading  mask %x", status);
+        return status;
+    }
+
+    ChipLogProgress(Zcl, "Refrigerator Alarm: Mask ep%d value: %" PRIu32 "", endpoint, mask->Raw());
+
+    return status;
+}
+
+EmberAfStatus RefrigeratorAlarmServer::GetStateValue(EndpointId endpoint, BitMask<AlarmMap> * state)
+{
+    EmberAfStatus status = Attributes::State::Get(endpoint, state);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        ChipLogProgress(Zcl, "Refrigerator Alarm: ERR: reading state %x", status);
+        return status;
+    }
+
+    ChipLogProgress(Zcl, "Refrigerator Alarm: State ep%d value: %" PRIu32 "", endpoint, state->Raw());
+
+    return status;
+}
+
+EmberAfStatus RefrigeratorAlarmServer::SetMaskValue(EndpointId endpoint, const BitMask<AlarmMap> & mask)
```

Comment:
One interesting question:  If the Mask changes to suppress a now-active alarm, what should happen to the State value?  Should it change?  @marktrayer

Reply:
Are we suppose to change the `State` when `Mask` changes? Spec says `State` cannot to change unless there is any physical action at the device ( closing the door ). Isn’t it conflicting ?

> the server allows the alarm to be remotely suppressed, and the modify command would suppress that alarm, the alarm should be reset.

Remotely alarm cannot be suppressed for this cluster.

Reply:
So when would Mask ever change for this cluster?  The answer is: only when physical action on the device suppresses the alarm, right?

And if physical action on the device suppresses the alarm, what should happen to State?

But more fundamentally: should we allow the cluster to get into a state where an alarm is:

1. Marked as suppressed in Mask.
2. Marked as currently alarming in State.

?  I would think we should not.  That means modifying State as needed when Mask changes, and perhaps rejecting State changes that don't match Mask or something...  @marktrayer @jadhavrohit924

Reply:
So there's two things to keep clean here I think:
1. the base cluster (assuming we have that implemented as a fundamental anywhere) supports Mask modification, and if you modify a mask to suppress an active alarm, then that alarm is suppressed and the state shall change.
2.  refrigerator alarm specifically doesn't support Mask modification, but if the alarm is physically suppressed locally (via some magic per device sequence), then state changes.

Reply:
> refrigerator alarm specifically doesn't support Mask modification

@marktrayer Including via physical interaction with the device?

Reply:
The note from @jadhavrohit924 is correct.  If the physical action suppresses the alarm, then the mask will change and the state will also change.  If subsequent physical action re-enables the alarm, the mask bit will be set.

Reply:
Changed state when mask changes